apps/aecore/src/txs/aec_spend_tx.erl

```diff
@@ -0,0 +1,152 @@
+-module(aec_spend_tx).
+
+%% API
+-export([new/2,
+         check/3,
+         process/3,
+         serialize/1,
+         deserialize/1,
+         type/0]).
+
+-behavior(aec_tx).
+
+-include("common.hrl").
+-include("trees.hrl").
+-include("txs.hrl").
+
+-spec new(map(), trees()) -> {ok, spend_tx()} | {error, sender_account_not_found}.
+new(#{sender := SenderPubkey,
+      recipient := RecipientPubkey,
+      amount := Amount,
+      fee := Fee}, Trees) ->
+    AccountsTrees = aec_trees:accounts(Trees),
+    case aec_accounts:get(SenderPubkey, AccountsTrees) of
+        {ok, #account{nonce = CurrentSenderNonce}} ->
+            {ok, #spend_tx{sender = SenderPubkey,
+                           recipient = RecipientPubkey,
+                           amount = Amount,
+                           fee = Fee,
+                           nonce = CurrentSenderNonce + 1}};
+        {error, notfound} ->
+            {error, sender_account_not_found}
+    end.
+
+-spec check(spend_tx(), trees(), height()) -> {ok, trees()} | {error, term()}.
+check(#spend_tx{recipient = RecipientPubkey} = SpendTx, Trees0, Height) ->
+    Checks = [fun check_tx_fee/3,
+              fun check_sender_account/3],
+    case aeu_validation:run(Checks, [SpendTx, Trees0, Height]) of
+        ok ->
+            case aec_tx_common:ensure_account_at_height(RecipientPubkey, Trees0, Height) of
+                {ok, Trees} ->
+                    {ok, Trees};
+                {error, account_height_too_big} ->
+                    {error, recipient_account_height_too_big}
+            end;
+        {error, _Reason} = Error ->
+            Error
+    end.
+
+-spec process(spend_tx(), trees(), height()) -> {ok, trees()}.
+process(#spend_tx{sender = SenderPubkey,
+                  recipient = RecipientPubkey,
+                  amount = Amount}, Trees0, Height) ->
+    AccountsTrees0 = aec_trees:accounts(Trees0),
+
+    {ok, SenderAccount0} = aec_accounts:get(SenderPubkey, AccountsTrees0),
+    {ok, RecipientAccount0} = aec_accounts:get(RecipientPubkey, AccountsTrees0),
+
+    {ok, SenderAccount} = aec_accounts:spend(SenderAccount0, Amount, Height),
+    {ok, RecipientAccount} = aec_accounts:earn(RecipientAccount0, Amount, Height),
+
+    {ok, AccountsTrees1} = aec_accounts:put(SenderAccount, AccountsTrees0),
+    {ok, AccountsTrees2} = aec_accounts:put(RecipientAccount, AccountsTrees1),
+
+    Trees = aec_trees:set_accounts(Trees0, AccountsTrees2),
+    {ok, Trees}.
+
+-spec serialize(spend_tx()) -> map().
+serialize(#spend_tx{sender = Sender,
+                    recipient = Recipient,
+                    amount = Amount,
+                    fee = Fee,
+                    nonce = Nonce}) ->
+    #{<<"sender">> => base64:encode(Sender),
```

Comment:
Please make this an array of maps instead, and without encoding:
```erlang
[#{<<"sender">> => Sender},
 #{<<"recipient">> => Recipient},
 ...]
```
C.f. https://github.com/aeternity/epoch/blob/gh-279-process-transactions/apps/aecore/src/txs/aec_coinbase_tx.erl#L54 (PR #310). Note also the inclusion of type and vsn.